plugins/inputs/influxdb_listener/http_listener.go

```diff
@@ -371,6 +380,16 @@ func (h *HTTPListener) parse(b []byte, t time.Time, precision string) error {
 	}
 
 	for _, m := range metrics {
+		// Do we need to keep the database name in the query string
+		if h.DatabaseTag != "" {
+			// Did we get a database argument. If we didn't get it. We can't set it.
+			if db != "" {
+				// Is there already a database set. If not use the database in the query string.
+				if _, ok := m.Tags()[h.DatabaseTag]; !ok {
```

Comment:
I think we should set the tag unconditionally, overwriting the tag from the line protocol:
```
if h.DatabaseTag != "" && db != "" {
    m.AddTag(h.DatabaseTag, db)
}
```

Reply:
Right, I guess this would make it act more like influxdb's write endpoint which is the point... I'll make this change and push it up.